source/opt/strip_debug_info_pass.cpp

```diff
@@ -22,10 +22,19 @@ Pass::Status StripDebugInfoPass::Process() {
   bool modified = !context()->debugs1().empty() ||
                   !context()->debugs2().empty() ||
                   !context()->debugs3().empty();
+  for (auto& dbg : context()->debugs1()) context()->KillNamesAndDecorates(&dbg);
+
+  for (auto& dbg : context()->debugs2()) context()->KillNamesAndDecorates(&dbg);
+
+  for (auto& dbg : context()->debugs3()) context()->KillNamesAndDecorates(&dbg);
+
   context()->debug_clear();
```

Comment:
If you call `KillInst` above, then this is not needed.  In fact, I would suggest removing `debug_clear` from the context and module classes.  It is a single purpose function and it does not keep the analyses valid, so they can be dangerous.

Reply:
Done